modules/apps/29-fee/keeper/keeper.go

```diff
@@ -112,43 +117,57 @@ func (k Keeper) EscrowAccountHasBalance(ctx sdk.Context, coins sdk.Coins) bool {
 // lockFeeModule sets a flag to determine if fee handling logic should run for the given channel
 // identified by channel and port identifiers.
 // Please see ADR 004 for more information.
-func (k Keeper) lockFeeModule(ctx sdk.Context) {
-	store := ctx.KVStore(k.storeKey)
-	store.Set(types.KeyLocked(), []byte{1})
+func (k Keeper) lockFeeModule(ctx context.Context) {
+	store := k.storeService.OpenKVStore(ctx)
+	if err := store.Set(types.KeyLocked(), []byte{1}); err != nil {
+		panic(err)
+	}
 }
 
 // IsLocked indicates if the fee module is locked
 // Please see ADR 004 for more information.
-func (k Keeper) IsLocked(ctx sdk.Context) bool {
-	store := ctx.KVStore(k.storeKey)
-	return store.Has(types.KeyLocked())
+func (k Keeper) IsLocked(ctx context.Context) bool {
+	store := k.storeService.OpenKVStore(ctx)
+	has, err := store.Has(types.KeyLocked())
+	if err != nil {
+		panic(err)
+	}
+	return has
 }
 
 // SetFeeEnabled sets a flag to determine if fee handling logic should run for the given channel
 // identified by channel and port identifiers.
-func (k Keeper) SetFeeEnabled(ctx sdk.Context, portID, channelID string) {
-	store := ctx.KVStore(k.storeKey)
-	store.Set(types.KeyFeeEnabled(portID, channelID), []byte{1})
+func (k Keeper) SetFeeEnabled(ctx context.Context, portID, channelID string) {
+	store := k.storeService.OpenKVStore(ctx)
+	if err := store.Set(types.KeyFeeEnabled(portID, channelID), []byte{1}); err != nil {
+		panic(err)
+	}
 }
 
 // DeleteFeeEnabled deletes the fee enabled flag for a given portID and channelID
-func (k Keeper) DeleteFeeEnabled(ctx sdk.Context, portID, channelID string) {
-	store := ctx.KVStore(k.storeKey)
-	store.Delete(types.KeyFeeEnabled(portID, channelID))
+func (k Keeper) DeleteFeeEnabled(ctx context.Context, portID, channelID string) {
+	store := k.storeService.OpenKVStore(ctx)
+	if err := store.Delete(types.KeyFeeEnabled(portID, channelID)); err != nil {
+		panic(err)
+	}
 }
 
 // IsFeeEnabled returns whether fee handling logic should be run for the given port. It will check the
 // fee enabled flag for the given port and channel identifiers
-func (k Keeper) IsFeeEnabled(ctx sdk.Context, portID, channelID string) bool {
-	store := ctx.KVStore(k.storeKey)
-	return store.Has(types.KeyFeeEnabled(portID, channelID))
+func (k Keeper) IsFeeEnabled(ctx context.Context, portID, channelID string) bool {
+	store := k.storeService.OpenKVStore(ctx)
+	has, err := store.Has(types.KeyFeeEnabled(portID, channelID))
```

Comment:
are the new `err` returns just propagations of errors that would surfaced before as panics? is there any risk here for actually getting an err back or would things need to be corrupted in the db somehow?

Reply:
there is also an error returned from `Get` now? how does that behave? 

if the key doesn't exist is there an error is it just `nil` bytes? Assuming the latter based on other code diffs

Reply:
we are now returning the panic we were doing internally. it is up to the caller to decide what to do with the error. Handling errors is a breaking change that i would like to avoid in these prs as panics to error handling can have hidden consequences.